### PR TITLE
fix: Log an error in development if URI size of 2000 characters is exceeded.

### PR DESCRIPTION
**Pull request by niklasbec**
As discussed [here](https://github.com/47ng/nuqs/pull/739#pullrequestreview-2416727212).

Error logging for when a user stores more than 2000 characters of state in a URI.

https://github.com/user-attachments/assets/c6f42fc3-c3e6-485e-b73f-39955ffe9f44

### CODE REVIEW

**Comment by niklasbec** [comment]
@franky47 wasn't to sure about the placement of the log, let me know.

**Review by franky47 — CHANGES_REQUESTED** [review]
Thanks! 

One global thing to change in addition to individual suggestions is the term URL rather than URI, to be consistent with the rest of the codebase (URLSearchParams, window.location etc).

File: packages/nuqs/src/update-queue.ts [hunk 1]
```diff
@@ -153,6 +153,9 @@ function flushUpdateQueue(
         scroll: options.scroll,
         shallow: options.shallow
       })
+      if (URIIsTooLong() && process.env.NODE_ENV === 'development') {
```
[the remark this answers is not in the document]
Comment (niklasbec) [reply packages/nuqs/src/update-queue.ts:156]: I know that Bun is designed with Node.js in mind so they expose the node env over process.env just like node does. AFAIK deno doesn't support this.

It works for vite and most other bundlers use plugins to expose process.env aswell.

[the remark this answers is not in the document]
Comment (niklasbec) [reply packages/nuqs/src/update-queue.ts:156]: We can remove the second check but then you would have to warnings in prod builds.

**Comment by niklasbec** [comment]
@franky47 I refactored according to your remarks, please have a look if this works for you.

**Review by franky47 — APPROVED** [review]

**Pull request merged**